packages/core/src/types.ts

```diff
@@ -515,6 +515,7 @@ export interface StateNodeDefinition<
   id: string;
   version: string | undefined;
   key: string;
+  context: TContext;
```

Comment:
This is not quite true, but probably hard to change in v4. When merging this to v5 we should remember about moving this onto root-related types.

Reply:
`.toJSON()` should only be called on the machine, anyway, where this will be defined.